Source/WebCore/rendering/RenderTheme.cpp

```diff
@@ -1284,7 +1288,8 @@ bool RenderTheme::isWindowActive(const RenderObject& renderer) const
 
 bool RenderTheme::isChecked(const RenderObject& o) const
 {
-    return is<HTMLInputElement>(o.node()) && downcast<HTMLInputElement>(*o.node()).shouldAppearChecked();
+    return (is<HTMLInputElement>(o.node()) && downcast<HTMLInputElement>(*o.node()).shouldAppearChecked())
+        || (o.node()->shadowHost() && is<HTMLInputElement>(o.node()->shadowHost()) && downcast<HTMLInputElement>(*o.node()->shadowHost()).shouldAppearChecked());
```

Comment:
This would be more readable as two separate statements, the use of `dynamicDowncast`, and an early return.